src/filters/blur_filter.class.ts

```diff
@@ -1,231 +1,201 @@
-//@ts-nocheck
-(function (global) {
-  'use strict';
-
-  var fabric = global.fabric || (global.fabric = {}),
-    filters = fabric.Image.filters,
-    createClass = fabric.util.createClass;
-
+// @ts-nocheck
+import { TClassProperties } from '../typedefs';
+import { createCanvasElement } from '../util/misc/dom';
+import { BaseFilter } from './base_filter.class';
+import {
+  TWebGLPipelineState,
+  T2DPipelineState,
+  TWebGLUniformLocationMap,
+  isWebGLPipelineState,
+} from './typedefs';
+
+/**
+ * Blur filter class
+ * @example
+ * const filter = new Blur({
+ *   blur: 0.5
+ * });
+ * object.filters.push(filter);
+ * object.applyFilters();
+ * canvas.renderAll();
+ */
+export class Blur extends BaseFilter {
   /**
-   * Blur filter class
-   * @class fabric.Image.filters.Blur
-   * @memberOf fabric.Image.filters
-   * @extends fabric.Image.filters.BaseFilter
-   * @see {@link fabric.Image.filters.Blur#initialize} for constructor definition
-   * @see {@link http://fabricjs.com/image-filters|ImageFilters demo}
-   * @example
-   * var filter = new fabric.Image.filters.Blur({
-   *   blur: 0.5
-   * });
-   * object.filters.push(filter);
-   * object.applyFilters();
-   * canvas.renderAll();
+   * blur value, in percentage of image dimensions.
+   * specific to keep the image blur constant at different resolutions
+   * range between 0 and 1.
+   * @type Number
+   * @default
    */
-  filters.Blur = createClass(
-    filters.BaseFilter,
-    /** @lends fabric.Image.filters.Blur.prototype */ {
-      type: 'Blur',
-
-      /*
-'gl_FragColor = vec4(0.0);',
```

Comment:
what is this?

Reply:
i don't remember. either a different blur or a more performant version of the same blur